src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1682,7 +1688,27 @@ static bool processSubscriptions
     }
 
     /* Check 2: expression (q) */
-    if (!matchExpression(notifyCerP, trigs->expression.q))
+    if (noCache == true)
```

Comment:
The pre-parsed string-filter is in the sub cache.
Without sub-cache, the string-filter will have to be parsed every time, with the 'q-string' obtained from mongo.

Now, if the parse is done here or at creation time of TriggeredSubscription  ... not so important, I'd say.